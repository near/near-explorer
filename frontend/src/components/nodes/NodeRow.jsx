import { Row, Col } from "react-bootstrap";

import Timer from "../utils/Timer";

export default class extends React.PureComponent {
  render() {
    const {
      nodeId,
      accountId,
      moniker,
      ipAddress,
      lastSeen,
      lastHeight,
      isValidator,
      agentName,
      agentVersion,
      agentBuild
    } = this.props;
    //TODO: Display on the list new information
    return (
      <Row className="node-row mx-0">
        <Col md="auto" xs="1" className="pr-0">
          <img
            src="/static/images/icon-m-block.svg"
            style={{ width: "15px" }}
          />
        </Col>
        <Col md="7" xs="6">
          <Row>
            <Col className="node-row-title">{moniker}</Col>
          </Row>
          <Row>
            <Col className="node-row-text">
              <Row>
                <Col md="auto">#{lastHeight}</Col>
                <Col md="auto" className="pl-0">
                  <img
                    src="/static/images/icon-m-user.svg"
                    style={{ width: "12px" }}
                  />
                  {` @${accountId}`}
                </Col>
                <Col md="auto">{ipAddress}</Col>
              </Row>
            </Col>
          </Row>
        </Col>
        <Col md="3" xs="4" className="ml-auto text-right">
          <Row>
            <Col className="node-row-txid" title={nodeId}>
              {nodeId.substring(8, 20)}...
            </Col>
          </Row>
          <Row>
            <Col className="node-row-timer">
              <span className="node-row-timer-status">Last seen</span>
              &nbsp;&nbsp;
              <Timer time={lastSeen} />
            </Col>
          </Row>
        </Col>
        <style jsx global>{`
          .node-row {
            padding-top: 10px;
            padding-bottom: 10px;
            border-top: solid 2px #f8f8f8;
          }

          .node-row:hover {
            background: rgba(0, 0, 0, 0.1);
          }

          .node-row-bottom {
            border-bottom: solid 2px #f8f8f8;
          }

          .node-row-title {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #24272a;
          }

          .node-row-text {
            font-family: BentonSans;
            font-size: 12px;
            font-weight: 500;
            line-height: 1.5;
            color: #999999;
          }

          .node-row-txid {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #0072ce;
          }

          .node-row-timer {
            font-family: BentonSans;
            font-size: 12px;
            color: #999999;
            font-weight: 100;
          }

          .node-row-timer-status {
            font-weight: 500;
          }
        `}</style>
      </Row>
    );
  }
}
