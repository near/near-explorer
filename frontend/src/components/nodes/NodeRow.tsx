import React from "react";
import { Row, Col } from "react-bootstrap";

import * as N from "../../libraries/explorer-wamp/nodes";

import Timer from "../utils/Timer";

interface Props {
  node: N.NodeInfo;
}

export default class extends React.PureComponent<Props> {
  statusIdentifier = new Map([
    ["AwaitingPeers", "Waiting for peers"],
    ["HeaderSync", "Syncing headers"],
    ["BlockSync", "Syncing blocks"],
    ["StateSync", "Syncing state"],
    ["StateSyncDone", "State sync is done"],
    ["BodySync", "Syncing body"],
    ["NoSync", ""],
  ]);
  render() {
    const { node } = this.props;
    return (
      <Row className="node-row mx-0">
        <Col md="auto" xs="1" className="pr-0">
          <img
            src={"/static/images/icon-m-node-online.svg"}
            style={{ width: "15px" }}
          />
        </Col>
        <Col md="7" xs="7">
          <Row>
            <Col className="node-row-title">
              @{node.accountId}{" "}
              <span className="node-status">
                {" "}
                {this.statusIdentifier.get(node.status)}
              </span>
            </Col>
          </Row>
          <Row>
            <Col className="node-row-text">
              <Row>
                <Col md={5}>
                  <img
                    src="/static/images/icon-m-size.svg"
                    style={{ width: "12px" }}
                  />
                  {`${node.agentName} | ver.${node.agentVersion} build  ${node.agentBuild}`}
                </Col>
                <Col>
                  <img
                    src="/static/images/icon-m-block.svg"
                    style={{ width: "12px" }}
                  />
                  {` ${node.lastHeight}`}
                </Col>
              </Row>
            </Col>
          </Row>
        </Col>
        <Col md="3" xs="3" className="ml-auto text-right">
          <Row>
            <Col className="node-row-txid" title={node.nodeId}>
              {node.nodeId.substring(8, 20)}...
            </Col>
          </Row>
          <Row>
            <Col className="node-row-timer">
              <span className="node-row-timer-status">Last seen</span>
              &nbsp;&nbsp;
              <Timer time={node.lastSeen} />
            </Col>
          </Row>
        </Col>
        <style jsx global>{`
          .node-row {
            padding-top: 10px;
            padding-bottom: 10px;
            border-top: solid 2px #f8f8f8;
          }

          .node-row:hover {
            background: rgba(0, 0, 0, 0.1);
          }

          .node-row-bottom {
            border-bottom: solid 2px #f8f8f8;
          }

          .node-row-title {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #24272a;
          }

          .node-row-text {
            font-family: BentonSans;
            font-size: 12px;
            font-weight: 500;
            line-height: 1.5;
            color: #999999;
          }

          .node-row-txid {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #4a4f54;
          }

          .node-row-timer {
            font-family: BentonSans;
            font-size: 12px;
            color: #999999;
            font-weight: 100;
          }

          .node-row-timer-status {
            font-weight: 500;
          }

          .node-status {
            font-size: 12px;
            line-height: 18px;
            color: #4a4f54;
          }
        `}</style>
      </Row>
    );
  }
}
