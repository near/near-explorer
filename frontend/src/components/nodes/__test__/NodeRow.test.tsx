import renderer from "react-test-renderer";

import NodeRow from "../NodeRow";

describe("<NodeRow />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <NodeRow
          node={{
            ipAddress: "1.1.1.1",
            moniker: "alice",
            accountId: "alice",
            nodeId: "ed25519:fhgjrkfhurjtieokwpitjesdlkfjngdmloytiutkjhgkf",
            lastSeen: 1587516758449,
            lastHeight: 1889,
            lastHash: "7a4d3SNNb1XGxqhHAKj5nGx8C1R4X8pdt8jobC9z3nup",
            signature:
              "ed25519:4n5VzXijHUvgwroD1W6BTysRGKTMdPv6ks32kEUNGyb9recDYRpt2L8Y2Uq5ABPt4LZHFBAv2DPj1CAPGkwqkZg2",
            agentName: "near-rs",
            agentVersion: "0.4.5",
            agentBuild: "6a527109-modified",
            peerCount: "0",
            isValidator: true,
            status: "NoSync",
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
