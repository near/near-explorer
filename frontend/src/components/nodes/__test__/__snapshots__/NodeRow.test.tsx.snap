// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NodeRow /> renders 1`] = `
<div
  className="node-row mx-0 row"
>
  <div
    className="pr-0 col-md-auto col-1"
  >
    <img
      className="jsx-990356129"
      src="/static/images/icon-m-node-online.svg"
      style={
        Object {
          "width": "15px",
        }
      }
    />
  </div>
  <div
    className="col-md-7 col-7"
  >
    <div
      className="row"
    >
      <div
        className="node-row-title col"
      >
        @
        alice
         
        <span
          className="jsx-990356129 node-status"
        >
           
          
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="node-row-text col"
      >
        <div
          className="row"
        >
          <div
            className="col-md-5"
          >
            <img
              className="jsx-990356129"
              src="/static/images/icon-m-size.svg"
              style={
                Object {
                  "width": "12px",
                }
              }
            />
            near-rs | ver.0.4.5 build  6a527109-modified
          </div>
          <div
            className="col"
          >
            <img
              className="jsx-990356129"
              src="/static/images/icon-m-block.svg"
              style={
                Object {
                  "width": "12px",
                }
              }
            />
             1889
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ml-auto text-right col-md-3 col-3"
  >
    <div
      className="row"
    >
      <div
        className="node-row-txid col"
        title="ed25519:fhgjrkfhurjtieokwpitjesdlkfjngdmloytiutkjhgkf"
      >
        fhgjrkfhurjt
        ...
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="node-row-timer col"
      >
        <span
          className="jsx-990356129 node-row-timer-status"
        >
          Last seen
        </span>
          
        <span>
          in 1y
        </span>
      </div>
    </div>
  </div>
</div>
`;
